node-graph/gcore/src/logic.rs

```diff
@@ -7,3 +7,38 @@ fn log_to_console<T: core::fmt::Debug>(value: T) -> T {
 	debug!("{:#?}", value);
 	value
 }
+
+pub struct LogicOrNode<Second> {
+	second: Second,
+}
+
+#[node_macro::node_fn(LogicOrNode)]
+fn logic_or(first: bool, second: bool) -> bool {
+	first || second
+}
+
+pub struct LogicAndNode<Second> {
+	second: Second,
+}
+
+#[node_macro::node_fn(LogicAndNode)]
+fn logic_and(first: bool, second: bool) -> bool {
+	first && second
+}
+
+pub struct LogicXorNode<Second> {
+	second: Second,
+}
+
+#[node_macro::node_fn(LogicXorNode)]
+fn logic_and(first: bool, second: bool) -> bool {
```

Comment:
The function name should be `logic_xor` (it doesn't matter as it is not used anywhere - just to reduce confusion).